### DISCUSSION TITLE
How to use AuthP with ISoftDelete

### DISCUSSION THREAD

**Discussion by MahmudX** [JonPSmith/AuthPermissions.AspNetCore#65] [Q&A]
EF core currently does not supports multiple query filters. However it can be done with `EfCore.SoftDeleteServices`.  My problem is I couldn't figure out how to use it with AuthP lib.

Can you guys give an example or demo to illustrate? 

It would be a great help.

Thank you.

**Comment by JonPSmith** [comment]
This is a `EfCore.SoftDeleteServices` issue, not a AuthP issue. You will found information in:

- The [Handling Query Filters with multiple filter parts](https://www.thereformedprogrammer.net/introducing-the-efcore-softdeleteservices-library-to-automate-soft-deletes/#handling-query-filters-with-multiple-filter-parts) section in the article "Introducing the EfCore.SoftDeleteServices library to automate soft deletes".
- The [Multiple Queryfilters](https://github.com/JonPSmith/EfCore.SoftDeleteServices/wiki/Multiple-Query-filters) documentation in the [EfCore.SoftDeleteServices](https://github.com/JonPSmith/EfCore.SoftDeleteServices) repo.

**Comment by MahmudX** [comment]
A few lines of code example would be helpful. I have read those articles but still I'm drifting middle of the ocean. :(

**Reply by JonPSmith** [reply]
The [EfCore.SoftDeleteServices](https://github.com/JonPSmith/EfCore.SoftDeleteServices) has a example of adding a soft delete query filter to a `DbContext` that also has another query filter, in this case called `userId` of type `GUID` - see [SingleSoftDelDbContext](https://github.com/JonPSmith/EfCore.SoftDeleteServices/blob/master/DataLayer/SingleEfCode/SingleSoftDelDbContext.cs).

In your case you would replace the `userId` with the AuthP `DataKey`. See [this unit test](https://github.com/JonPSmith/EfCore.SoftDeleteServices/blob/master/Test/UnitTests/SingleSoftDeleteTests/TestSetSoftDelete.cs#L77L98) for how that works.
